internal/bminventory/inventory.go

```diff
@@ -882,6 +886,32 @@ func (b *bareMetalInventory) createAndUploadDay2NodeIgnition(ctx context.Context
 	return nil
 }
 
+func (b *bareMetalInventory) UpdateIgnitionEndpointIfHasMCSCert(log logrus.FieldLogger, host *models.Host, cluster *common.Cluster, ignitionEndpointUrl string) (string, error) {
+	// No ignition override set by bmh agent controller
+	if host.IgnitionConfigOverrides == "" {
+		return ignitionEndpointUrl, nil
+	}
+	// No CA certs set in ignition override
+	if !ignition.HasCACertInIgnition(host.IgnitionConfigOverrides) {
+		return ignitionEndpointUrl, nil
+	}
+	// Not an operator-backed deployment
+	if b.k8sClient == nil {
+		return ignitionEndpointUrl, nil
+	}
+	log.Infof("Found CA certificate in ignition override for cluster %s, host %s", cluster.ID, host.ID)
+	url, err := url.Parse(ignitionEndpointUrl)
+	if err != nil {
+		return ignitionEndpointUrl, err
+	}
+	url.Scheme = "https"
+	apiIntHostname := fmt.Sprintf("%s.%s.%s", constants.InternalAPIClusterSubdomain, cluster.Name, cluster.BaseDNSDomain)
+	url.Host = fmt.Sprintf("%s:22623", apiIntHostname)
```

Comment:
Do we have this format and port somewhere else? Seems like it would be a good thing to share.

Reply:
Also I don't see why this needs to be two `Sprintf` calls.

Reply:
Moved ports to `constants`, removed intermediate variable